### PR TITLE
fixes and enchancements for contrib/hadoop

### PR DESCRIPTION
**Pull request by nevillelyh**

### CODE REVIEW

**Comment by davorbonaci** [comment]
@tomwhite Tom, as an original writer, do you have any comments on this pull request? If not, we'll do a quick review and merge @nevillelyh's changes.

File: contrib/hadoop/src/main/java/com/google/cloud/dataflow/contrib/hadoop/AvroHadoopFileSource.java [hunk 1]
```diff
@@ -0,0 +1,86 @@
+package com.google.cloud.dataflow.contrib.hadoop;
```
Comment (davorbonaci) [inline contrib/hadoop/src/main/java/com/google/cloud/dataflow/contrib/hadoop/AvroHadoopFileSource.java]: We require a copyright header in every file. Could you add it please?

File: contrib/hadoop/src/main/java/com/google/cloud/dataflow/contrib/hadoop/AvroWrapperCoder.java [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+
+  @Override
+  public CloudObject asCloudObject() {
+    CloudObject result = super.asCloudObject();
+    result.put("wrapperType", wrapperType.getName());
+    return result;
+  }
+
+  @Override
+  public void verifyDeterministic() throws NonDeterministicException {}
```
Comment (davorbonaci) [inline contrib/hadoop/src/main/java/com/google/cloud/dataflow/contrib/hadoop/AvroWrapperCoder.java]: Is this really a deterministic coder?

File: contrib/hadoop/src/main/java/com/google/cloud/dataflow/contrib/hadoop/HadoopFileSink.java [hunk 3]
```diff
@@ -0,0 +1,211 @@
+package com.google.cloud.dataflow.contrib.hadoop;
```
Comment (davorbonaci) [inline contrib/hadoop/src/main/java/com/google/cloud/dataflow/contrib/hadoop/HadoopFileSink.java]: Copyright header is needed here.

**Comment by davorbonaci** [comment]
I put in a few comments. The main comment would be to add more tests. It is often easy to test coders using `CoderProperties` utility. That said, we probably wouldn't block merging this pull request on it.

**Comment by nevillelyh** [comment]
@davorbonaci I addressed most of your comments. The only thing missing is tests for the new source/sink.

**Comment by davorbonaci** [comment]
@nevillelyh We are about ready to take this pull request, but we'd like to take it directly in our Apache Beam repository. Would you mind resubmitting there?

Note pending apache/incubator-beam#96, which does some renaming. If possible, please apply this fix on top of that change.

And, thanks for this! Much appreciated.

**Pull request closed without merging**